ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -95,21 +107,30 @@ class MySqlNamedBlobDb implements NamedBlobDb {
   private static final String LIST_QUERY = String.format("SELECT %1$s, %2$s, %3$s, %4$s FROM %5$s "
           + "WHERE (%6$s, %7$s) = (?, ?) AND %1$s LIKE ? AND %1$s >= ? ORDER BY %1$s ASC LIMIT ?", BLOB_NAME, BLOB_ID,
       EXPIRES_TS, DELETED_TS, NAMED_BLOBS, ACCOUNT_ID, CONTAINER_ID);
+  private static final String LIST_QUERY_V2 = String.format("SELECT t1.blob_name, t1.blob_id, t1.version, t1.deleted_ts "
+          + "FROM named_blobs_v2 t1 INNER JOIN (SELECT account_id, container_id, blob_name, max(version) as version FROM named_blobs_v2 "
+          + "WHERE (account_id, container_id) = (?, ?) AND %1$s GROUP BY account_id, container_id, blob_name) t2 ON "
+          + "(t1.account_id,t1.container_id,t1.blob_name,t1.version) = (t2.account_id,t2.container_id,t2.blob_name,t2.version) "
+          + "WHERE t1.blob_name LIKE ? AND t1.blob_name >= ? ORDER BY t1.blob_name ASC LIMIT ?",
+      STATE_MATCH);
```

Comment:
is this really necessary? I am not an expert on sql statement, but it seems like we can do something like
```
SELECT blob_name, blob_id, max(version), deleted_ts FROM named_blobs_v2 WHERE (account_id, container_id) = (?, ?) AND blob_state = Ready AND  blob_name LIKE ? AND blob_name >= ?  GROUP BY account_id, container_id, blob_name ORDER BY blob_name ASC LIMIT ?
```
Does this look right to you ?

Reply:
No, your query has problem because:
"All column names in SELECT list must appear in GROUP BY clause unless name is used only in an aggregate function" https://stackoverflow.com/questions/5986127/do-all-columns-in-a-select-list-have-to-appear-in-a-group-by-clause

I actually spent quite some time to come up with above query, it looks long, but the idea is simple: 

```
1. Get the max version per each (account_id, container_id, blob_name) as t2.
2. Join the original table with t2 on the primary keys to filter out historical data with older versions.
3. And then we can apply the blob_name like ?, and >, order by and limit per page size.
```